pkg/loop/internal/chain_reader_test.go

```diff
@@ -0,0 +1,235 @@
+
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb"
+
+	"github.com/fxamacker/cbor/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	. "github.com/smartcontractkit/chainlink-common/pkg/types/interfacetests"
+)
+
+func TestVersionedBytesFunctions(t *testing.T) {
+	const unsupportedVer = 25913
+	t.Run("EncodeVersionedBytes unsupported type", func(t *testing.T) {
+		expected := types.InvalidTypeError
+		invalidData := make(chan int)
+
+		_, err := encodeVersionedBytes(invalidData, JSONEncodingVersion2)
+		if err == nil || !errors.Is(err, expected) {
+			t.Errorf("expected error: %s, but got: %v", expected, err)
+		}
+	})
+
+	t.Run("EncodeVersionedBytes unsupported encoding version", func(t *testing.T) {
+		expected := fmt.Errorf("unsupported encoding version %d for data map[key:value]", unsupportedVer)
+		data := map[string]interface{}{
+			"key": "value",
+		}
+
+		_, err := encodeVersionedBytes(data, unsupportedVer)
+		if err == nil || err.Error() != expected.Error() {
+			t.Errorf("expected error: %s, but got: %v", expected, err)
+		}
+	})
+
+	t.Run("DecodeVersionedBytes", func(t *testing.T) {
+		var decodedData map[string]interface{}
+		expected := fmt.Errorf("unsupported encoding version %d for versionedData [97 98 99 100 102]", unsupportedVer)
+		versionedBytes := &pb.VersionedBytes{
+			Version: unsupportedVer, // Unsupported version
+			Data:    []byte("abcdf"),
+		}
+
+		err := decodeVersionedBytes(&decodedData, versionedBytes)
+		if err == nil || err.Error() != expected.Error() {
+			t.Errorf("expected error: %s, but got: %v", expected, err)
+		}
+	})
+}
+
+func TestChainReaderClient(t *testing.T) {
+	RunChainReaderInterfaceTests(t, &interfaceTester{})
+
+	lis := bufconn.Listen(1024 * 1024)
+	s := grpc.NewServer()
+	es := &errorServer{}
+	pb.RegisterChainReaderServer(s, es)
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			panic(err)
+		}
+	}()
+	defer s.Stop()
+	conn := connFromLis(t, lis)
+	client := &chainReaderClient{grpc: pb.NewChainReaderClient(conn)}
+	ctx := context.Background()
+
+	errorTypes := []error{
+		types.InvalidTypeError,
+		types.FieldNotFoundError,
+	}
+
+	for _, errorType := range errorTypes {
+		es.err = errorType
+		t.Run("GetLatestValue unwraps errors from server "+errorType.Error(), func(t *testing.T) {
+			err := client.GetLatestValue(ctx, types.BoundContract{}, "method", "anything", "anything")
+			assert.IsType(t, errorType, err)
+		})
+	}
+
+	// make sure that errors come from client directly
+	es.err = nil
+	invalidTypeErr := types.InvalidTypeError
+	t.Run("GetLatestValue returns error if type cannot be encoded in the wire format", func(t *testing.T) {
+		err := client.GetLatestValue(ctx, types.BoundContract{}, "method", &cannotEncode{}, &TestStruct{})
+		assert.NotNil(t, errors.As(err, &invalidTypeErr))
+	})
+}
+
+type interfaceTester struct {
+	lis    *bufconn.Listener
+	server *grpc.Server
```

Comment:
Is this meant to hold the `s` server from `Setup`?

Reply:
Looks like it to me, updating